lib/index.js

```diff
@@ -86,26 +93,35 @@ module.exports = {
   convert (language, variant, request, options, callback) {
     var convert, main;
 
-    if (!sdk.Request.isRequest(request)) {
-      return callback('Codegen~convert: Invalid request');
-    }
+    try {
+      if (!sdk.Request.isRequest(request)) {
+        throw TypeError('Codegen~convert: Invalid request');
+      } 
 
-    languageMap.forEach((codegen) => {
-      var lang = codegen.lang.trim(),
-        currentVariant = codegen.variant.trim();
-      if (language.toLowerCase() === lang.toLowerCase() && variant.toLowerCase() === currentVariant.toLowerCase()) {
-        main = codegen.main;
-        convert = main.convert;
-
-        if (typeof convert !== 'function') {
-          return callback('Codegen~convert: Convert is not a function');
+      languageMap.forEach((codegen) => {
+        var lang = codegen.lang.trim(),
+          currentVariant = codegen.variant.trim();
+        if (language.toLowerCase() === lang.toLowerCase() && variant.toLowerCase() === currentVariant.toLowerCase()) {
+          main = codegen.main;
+          convert = main.convert;
+  
+          if (typeof convert !== 'function') {
+            throw TypeError('Codegen~convert: Convert is not a function');
+          }
         }
+      });
+
+      if (!convert) {
+        throw ReferenceError('Codegen~convert: Could not find condegen corresponding to provided language, variant pair');
       }
-    });
-    if (!convert) {
-      return callback('Codegen~convert: Could not find condegen corresponding to provided language, variant pair');
-    }
 
+    } catch(e) {
+      if(e instanceof TypeError || e instanceof ReferenceError) {
+        console.log(e.message);
```

Comment:
Same comment. 
Just by logging the error, the main issue is not solved. 
You need to return from inside the catch block itself.